src/compression.cpp

```diff
@@ -26,22 +26,29 @@
 #include <stdexcept>
 
 const std::string LZMA_INFO::name = "lzma";
-void LZMA_INFO::init_stream_decoder(stream_t* stream, char* raw_data)
+
+void LZMA_INFO::init_stream_encoder(stream_t* stream, int compression_level, char* raw_data)
 {
   *stream = LZMA_STREAM_INIT;
-  unsigned memsize = zim::envMemSize("ZIM_LZMA_MEMORY_SIZE", LZMA_MEMORY_SIZE * 1024 * 1024);
-  auto errcode = lzma_stream_decoder(stream, memsize, 0);
+  int cl = compression_level;
+
+  if (cl == static_cast<int>(zim::LZMACompressionLevel::MAXIMUM)) {
+    cl |= LZMA_PRESET_EXTREME;
+  }
```

Comment:
We should not change the compression level passed by the user. It should be a simple pass through.
With this, a user would never be able to set a compression level of 9 without `LZMA_PRESET_EXTREME`.